modules/subschema/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/subschema/package.scala

```diff
@@ -12,23 +14,34 @@ package object subschema {
 
   def canonicalize(s: Schema): Schema =
     s match {
-      case s if s.`type` == Some(Integer) => s.copy(`type` = Some(Number))
-      case s if s.`type` == Some(Number)  => s
-      case s if s.`type` == Some(String)  => s
+      case s if s.`type`.contains(Null)    => s
+      case s if s.`type`.contains(Boolean) =>
+        if (s.`enum`.isDefined) s else s.copy(`enum` = Some(Enum(List(Json.True, Json.False))))
```

Comment:
this line assumes that type is defined, the case where type is not defined but enum is present will be dealt with in the recursive case - to be implemented later - that deals with heterogeneous arrays. 
If enum is defined too, then no op (if defined it will contain at least one element otherwise the schema would fail validation), if it's not defined we make the possible values explicit by defining enum as the collection of the only two possible values of a boolean type